corehq/apps/ota/views.py

```diff
@@ -32,6 +42,69 @@ def restore(request, domain, app_id=None):
     return get_restore_response(domain, couch_user, app_id, **get_restore_params(request))
 
 
+@json_error
+@login_or_digest_or_basic_or_apikey()
+def search(request, domain):
+    """
+    Accepts search criteria as GET params, e.g. "https://www.commcarehq.org/a/domain/phone/search/?a=b&c=d"
+    Returns results as a fixture with the same structure as a casedb instance.
+    """
+    criteria = dict(request.GET)
+    try:
+        case_type = criteria.pop('case_type')
+    except KeyError:
+        return HttpResponse('Search request must specify case type', status=400)
+
+    search_es = CaseSearchES()
+    search_es = search_es.domain(domain)
+    search_es = search_es.case_type(case_type)
+    config = CaseSearchConfig(domain=domain).config
+    fuzzies = config.get_fuzzy_properties_for_case_type(case_type)
+    for key, value in criteria.items():
+        search_es = search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+    results = search_es.values()
+    if should_use_sql_backend(domain):
+        fixtures = CaseDBFixture([get_instance_from_data(CommCareCaseSQLSerializer, result) for result in results])
+    else:
+        fixtures = CaseDBFixture([CommCareCase.wrap(result) for result in results])
+    return HttpResponse(fixtures, content_type="text/xml")
+
+
+@json_error
+@login_or_digest_or_basic_or_apikey()
+def claim(request, domain):
+    couch_user = CouchUser.from_django_user(request.user)
+    # TODO: use request.session to ensure only one extension case if multiple claims for the same beneficiary
+    #       come from the same AWW.
+    if request.method == 'POST':
+        host_id = request.POST['case_id']
+        host_type = request.POST.get('case_type')  # Nice to have,
+        host_name = request.POST.get('case_name')  # but optional
+        if not (host_type and host_name):
+            if should_use_sql_backend(domain):
+                case = CommCareCaseSQL.objects.get(case_id=host_id)
+            else:
+                case = CommCareCase.get(host_id)
```

Comment:
Can replace the if else with this `case = CaseAccessors(domain).get_case(host_id)`